Automattic/jetpack
# Jetpack Boost: Improve error handling and response consistency

## Proposed changes:
- Refactor DataSyncError and related DataSyncHooks to handle error states more consistently and provide utility methods like isAborted.
- Implement response formatting functions in PHP action and endpoint classes for consistent error and success responses.
- Simplify page cache setup and error handling in the Jetpack Boost plugin by moving cache setup calls and removing redundant hooks.
- Standardize error handling across the plugin by using the DataSyncError.info method and standardize-error utility.

### Other information:



## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
N/A

## Testing instructions:
- Trigger error states in the DataSync client and verify that the 'isAborted' method correctly identifies aborted requests.
- Invoke actions in the PHP classes that result in errors or success and ensure that the response formats are consistent with the new methods.
- Enable and disable page cache in Jetpack Boost to test if the setup and error handling works without issues.
- Check various components that handle errors to confirm that the standardized error messages are displayed correctly.
